File: audiorelay_flutter/lib/models/server_info.dart
/// Information about a discovered AudioRelay server.
class ServerInfo {
  final String id;
  final String name;
  final String host;
  final int port;
  final int audioPort;
  final String version;
  final List<String> features;
  final String? osVersion;

  const ServerInfo({
    required this.id,
    required this.name,
    required this.host,
    this.port = 59200,
    this.audioPort = 59100,
    this.version = '0.27.5',
    this.features = const [],
    this.osVersion,
  });

  factory ServerInfo.fromJson(Map<String, dynamic> json) {
    return ServerInfo(
      id: json['id'] as String? ?? '${json['host']}:${json['port']}',
      name: json['name'] as String? ?? 'AudioRelay Server',
      host: json['host'] as String? ?? 'unknown',
      port: json['port'] as int? ?? 59200,
      audioPort: json['audioPort'] as int? ?? 59100,
      version: json['version'] as String? ?? '0.27.5',
    );
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) || other is ServerInfo && id == other.id;

  @override
  int get hashCode => id.hashCode;
}

/// Audio format configuration.
class AudioFormat {
  final int sampleRate;
  final int channels;
  final int bitsPerSample;
  final int bitrate;

  const AudioFormat({
    this.sampleRate = 48000,
    this.channels = 2,
    this.bitsPerSample = 16,
    this.bitrate = 96000,
  });
}

/// Network statistics.
class NetworkStats {
  int bytesReceived = 0;
  int packetsReceived = 0;
  int packetsLost = 0;
  double averageLatency = 0;
  double maxLatency = 0;
  int underflowCount = 0;
  bool hasOutOfOrderPackets = false;
}

File: AudioRelayClient/Sources/Model/ServerInfo.swift
import Foundation

/// Information about a discovered AudioRelay server
struct ServerInfo: Identifiable, Codable {
    let id: String
    let name: String
    let host: String
    let port: Int
    let audioPort: Int
    let version: String
    let features: [String]
    let osVersion: String?

    init(
        id: String = UUID().uuidString,
        name: String,
        host: String,
        port: Int = 59200,
        audioPort: Int = 59100,
        version: String = "0.27.5",
        features: [String] = [],
        osVersion: String? = nil
    ) {
        self.id = id
        self.name = name
        self.host = host
        self.port = port
        self.audioPort = audioPort
        self.version = version
        self.features = features
        self.osVersion = osVersion
    }
}

/// Audio format configuration
struct AudioFormat: Equatable {
    let sampleRate: Int
    let channels: Int
    let bitsPerSample: Int
    let bitrate: Int

    static let defaultFormat = AudioFormat(
        sampleRate: 48000,
        channels: 2,
        bitsPerSample: 16,
        bitrate: 96000
    )
}

/// Network statistics
struct NetworkStats {
    var bytesReceived: UInt64 = 0
    var packetsReceived: UInt64 = 0
    var packetsLost: UInt64 = 0
    var averageLatency: Double = 0
    var maxLatency: Double = 0
    var underflowCount: UInt64 = 0
    var outOfOrderPackets: UInt64 = 0
}
